src/Layouts/Layout.php

```diff
@@ -209,74 +223,107 @@ public function get()
     {
         $this->applyFilters();
         $this->applySort();
-        $response = [
-            'data' => $this->convertToArray($this->command->execute()->getRecords())
-        ];
-        $this->attachPaginationData($response);
+        $rawResponse = $this->command->execute();
+        if ($this->pagination){
+            $response = $this->createPaginationData($rawResponse);
+        }
+        $response ['data'] = $this->convertToArray($this->filemaker, $rawResponse->getRecords());
         return $response;
     }
 
+    //      SORTING     //
     /**
-     * executes the insert command and returns mapped array of the created object or throws exception
+     * add sorting rule asc or desc
      *
-     * @return array
-     * @throws FieldDoesNotExist
-     * @throws FileMakerException
+     * @param $key
+     * @param $order
+     * @return $this
      */
-    public function post()
+    public function sort($key, $order = 'asc')
     {
-        $response = [
-            'data' => $this->convertToArray($this->command->execute()->getRecords())
-        ];
-        return $response;
+        $order = $order == 'desc'? FileMaker::SORT_DESCEND : FileMaker::SORT_ASCEND;
+        $this->sortRule = [$key => $order];
+        return $this;
     }
 
     /**
-     * executes the edit command and returns  mapped array of the updated object or throws exception
+     * apply sort rule
+     */
+    public function applySort()
+    {
+        $this->command->addSortRule(key($this->sortRule), 1, current($this->sortRule));
+    }
+    //      END SORTING     //
+
+    //      Filtering     //
+    /**
+     * adds filters to the query
      *
-     * @return array
-     * @throws FieldDoesNotExist
-     * @throws FileMakerException
+     * @param $filters
+     * @param null $value
+     * @return $this
      */
-    public function update()
+    public function filter($filters, $value = null)
     {
-        $response = [
-            'data' => $this->convertToArray($this->command->execute()->getRecords())
-        ];
-        return $response;
+        if (is_string($filters))
```

Comment:
I am not sure what is going on here can we clean this up a little bit may be force a type-hint that this would be passed as and handled in the controller this seems quite ambiguous what it does. I think we should define the api to pass the filters variable as and whether it's a string or an array this needs to be handled on the consumers side and this method should always receive the filters as an array.